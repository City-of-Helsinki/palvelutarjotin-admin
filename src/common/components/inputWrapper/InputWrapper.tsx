import classNames from 'classnames';
import React from 'react';

import FieldLabel from '../fieldLabel/FieldLabel';
import styles from './inputWrapper.module.scss';

export type InputWrapperProps = {
  children?: React.ReactNode;
  className?: string;
  errorText?: string;
  hasIcon?: boolean;
  helperText?: string;
  hideLabel?: boolean;
  id: string;
  invalid?: boolean;
  label?: string | React.ReactNode;
  required?: boolean;
  style?: React.CSSProperties;
  successText?: string;
  tooltipLabel?: string;
  tooltipText?: string;
  tooltipButtonLabel?: string;
};

const InputWrapper: React.FC<InputWrapperProps> = ({
  children,
  className = '',
  errorText,
  hasIcon = false,
  helperText,
  hideLabel = false,
  id,
  invalid = false,
  label,
  required = false,
  style,
  successText,
  tooltipLabel,
  tooltipText,
  tooltipButtonLabel,
}) => (
  <div
    className={classNames(
      styles.root,
      {
        [styles.hasIcon]: hasIcon,
        [styles.invalid]: invalid,
        [styles.success]: successText,
      },
      className
    )}
    style={style}
  >
    {label && (
      <FieldLabel
        inputId={id}
        hidden={hideLabel}
        label={label}
        required={required}
        tooltipLabel={tooltipLabel}
        tooltipButtonLabel={tooltipButtonLabel}
        tooltipText={tooltipText}
      />
    )}
    <div className={classNames(styles.inputWrapper)}>{children}</div>
    {errorText && (
      <div className={styles.errorText} id={`${id}-error`}>
        {errorText}
      </div>
    )}
    {successText && (
      <div className={styles.successText} id={`${id}-success`}>
        {successText}
      </div>
    )}
    {helperText && (
      <div className={styles.helperText} id={`${id}-helper`}>
        {helperText}
      </div>
    )}
  </div>
);

export default InputWrapper;
