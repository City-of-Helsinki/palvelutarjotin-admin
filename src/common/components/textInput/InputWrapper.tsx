import classNames from 'classnames';
import { Tooltip } from 'hds-react';
import React, { CSSProperties, FC, ReactNode } from 'react';

import styles from './inputWrapper.module.scss';

export type InputWrapperProps = {
  children?: ReactNode;
  className?: string;
  hasIcon?: boolean;
  helperText?: string;
  hideLabel?: boolean;
  id: string;
  invalid?: boolean;
  labelId?: string;
  labelText?: string;
  style?: CSSProperties;
  tooltipLabel?: string;
  tooltipText?: string;
  tooltipOpenButtonLabelText?: string;
  tooltipCloseButtonLabelText?: string;
  required?: boolean;
};

const InputWrapper: FC<InputWrapperProps> = React.forwardRef<
  HTMLDivElement,
  InputWrapperProps
>(
  (
    {
      children,
      className = '',
      hasIcon = false,
      helperText,
      hideLabel = false,
      id,
      invalid = false,
      labelId,
      labelText,
      style,
      tooltipLabel,
      tooltipText,
      tooltipOpenButtonLabelText,
      tooltipCloseButtonLabelText,
      required,
    },
    ref
  ) => (
    <div
      className={classNames(
        styles.root,
        hasIcon && styles.hasIcon,
        invalid && styles.invalid,
        className
      )}
      ref={ref}
      style={style}
    >
      {labelText && (
        <label
          id={labelId}
          htmlFor={id}
          className={`${styles.label} ${hideLabel ? styles.hiddenLabel : ''}`}
        >
          {labelText}
          {required && <span className={styles.requiredIndicator}>*</span>}
        </label>
      )}
      {tooltipText && (
        <Tooltip
          tooltipLabel={tooltipLabel || ''}
          // closeButtonLabelText={tooltipCloseButtonLabelText || ''}
          buttonLabel={tooltipOpenButtonLabelText || ''}
        >
          {tooltipText}
        </Tooltip>
      )}
      <div className={classNames(styles.inputWrapper)}>{children}</div>
      {helperText && <div className={styles.helperText}>{helperText}</div>}
    </div>
  )
);

export default InputWrapper;
