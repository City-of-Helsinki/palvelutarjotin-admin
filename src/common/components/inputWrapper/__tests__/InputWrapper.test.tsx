import { render } from '@testing-library/react';
import React from 'react';

import InputWrapper, { InputWrapperProps } from '../InputWrapper';

const wrapperProps: InputWrapperProps = {
  helperText: 'helper text',
  label: 'label text',
  id: 'test',
  successText: 'Success',
};

it('matched snapshot', () => {
  const { asFragment } = render(
    <InputWrapper {...wrapperProps}>
      <input id="test" />
    </InputWrapper>
  );
  expect(asFragment()).toMatchSnapshot();
});
