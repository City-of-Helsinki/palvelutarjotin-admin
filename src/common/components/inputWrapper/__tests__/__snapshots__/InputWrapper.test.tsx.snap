// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<DocumentFragment>
  <div
    class="root success"
  >
    <label
      class="label "
      for="test"
    >
      label text
    </label>
    <div
      class="inputWrapper"
    >
      <input
        id="test"
      />
    </div>
    <div
      class="successText"
      id="test-success"
    >
      Success
    </div>
    <div
      class="helperText"
      id="test-helper"
    >
      helper text
    </div>
  </div>
</DocumentFragment>
`;
